### ISSUE TITLE
avatar updates

### ISSUE THREAD

**Issue by DarkMatters** [VarynInc/varyn#176]
setup defaults for
+ male
+ female
+ unknown (monster)

order:
1. browser cache
2. avatar already composited in db
3. avatar composite defined assets selected
4. users.img_url
5. default gender composite
6. default file

Unit tests:
+ [x] default male
+ [x] default female
+ [x] default unknown
+ [x] Enginesis user male
+ [x] Enginesis user female
+ [x] Enginesis user cached
+ [x] Facebook user
+ [x] Twitteruser
+ [x] Google user

+ [x] Local Enginesis default male: http://www.enginesis-l.com/avatar.php?site_id=100&default_index=0&size=1
+ [x] Local Enginesis default female: http://www.enginesis-l.com/avatar.php?site_id=100&default_index=1&size=1
+ [x] Local Enginesis default monster: http://www.enginesis-l.com/avatar.php?site_id=100&default_index=2&size=1
+ [x] Local Enginesis user without avatar (get default for Gender): http://www.enginesis-l.com/avatar.php?site_id=100&user_id=10243&size=1
+ [ ] Local Varyn user with avatar: http://www.enginesis-l.com/avatar.php?site_id=106&user_id=10239&size=1
+ [x] Local Google (7): 10567 http://www.enginesis-l.com/avatar.php?site_id=106&user_id=10567&size=1  https://lh3.googleusercontent.com/-VxZgZkhnb74/AAAAAAAAAAI/AAAAAAAAB14/LXZ5OFRkWlQ/s72-p-k-rw-no/photo.jpg
+ [x] Local Facebook (2): 10566 http://www.enginesis-l.com/avatar.php?site_id=106&user_id=10566&size=1 http://graph.facebook.com/100003309700579/picture?type=square
+ [x] Local Twitter (11): 10565 http://www.enginesis-l.com/avatar.php?site_id=106&user_id=10565&size=1 https://pbs.twimg.com/profile_images/677586306365394944/UmdV4vTi_bigger.jpg
---
+ [x] -q Varyn default male: http://www.enginesis-q.com/avatar.php?site_id=106&default_index=0&size=1
+ [x] -q Varyn default female: http://www.enginesis-q.com/avatar.php?site_id=106&default_index=1&size=1
+ [x] -q Varyn default monster: http://www.enginesis-q.com/avatar.php?site_id=106&default_index=2&size=1
+ [x] -q Varyn user without avatar (get default for Gender user 10553 is F, 10554 is U, 10555 in M): http://www.enginesis-q.com/avatar.php?site_id=106&user_id=10553&size=1
+ [x] -q Varyn user without avatar (get default for Gender user 10246 is M): http://www.enginesis-q.com/avatar.php?site_id=106&user_id=10246&size=1
+ [x] -q Enginesis user with avatar: http://www.enginesis-q.com/avatar.php?site_id=106&user_id=10239&size=1
+ [x] -q Google(7): 10560 http://www.enginesis-q.com/avatar.php?site_id=106&user_id=10567&size=1  https://lh3.googleusercontent.com/-VxZgZkhnb74/AAAAAAAAAAI/AAAAAAAAB14/LXZ5OFRkWlQ/s72-p-k-rw-no/photo.jpg
+ [x] -q Facebook(2): 10566 http://www.enginesis-q.com/avatar.php?site_id=106&user_id=10566&size=1 http://graph.facebook.com/100003309700579/picture?type=square
+ [x] -q Twitter(11): 10565 http://www.enginesis-q.com/avatar.php?site_id=106&user_id=10565&size=1 https://pbs.twimg.com/profile_images/677586306365394944/UmdV4vTi_bigger.jpg

Depends on #177 #175

**Comment by varyndev** [comment]
fully coded requires testing on live to wire up with co-reg.

**Comment by varyndev** [comment]
5 tests are failing on -q (see 1st comment unchecked tests)

**Comment by varyndev** [comment]
completed tests, working per issue description. we need to make several improvements so open a new issue(s) for those.
